### PR TITLE
Resolve numerous Coverity findings.

### PR DESCRIPTION
**Pull request by sjudeng**
Relates to #535

### CODE REVIEW

File: janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/util/JanusGraphVertexDeserializer.java [hunk 1]
```diff
@@ -219,10 +219,7 @@ public TinkerVertex readHadoopVertex(final StaticBuffer key, Iterable<Entry> ent
 
         /*Since we are filtering out system relation types, we might end up with vertices that have no incident relations.
          This is especially true for schema vertices. Those are filtered out.     */
-        if (!foundVertexState) {
-            log.trace("Vertex {} has unknown lifecycle state", vertexId);
-            return null;
-        } else if (!tv.edges(Direction.BOTH).hasNext() && !tv.properties().hasNext()) {
+        if (!tv.edges(Direction.BOTH).hasNext() && !tv.properties().hasNext()) {
```
Comment (davidclement90) [inline janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/util/JanusGraphVertexDeserializer.java]: Why did you remove the log ?
And can you delete the dead code just above ?

Comment (sjudeng) [reply janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/util/JanusGraphVertexDeserializer.java]: @davidclement90 This section is unreachable because it's conditioned on `!foundVertexState` and `foundVertexState` is set to `!verifyVertexExistence` ([L93](https://github.com/JanusGraph/janusgraph/blob/84ead7d907b62061e30c5d3210e5e8422fd4648c/janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/util/JanusGraphVertexDeserializer.java#L93)), which is always true since `verifyVertexExistence` is `false` ([L48](https://github.com/JanusGraph/janusgraph/blob/84ead7d907b62061e30c5d3210e5e8422fd4648c/janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/util/JanusGraphVertexDeserializer.java#L48)). I never would have found this if it wasn't for Coverity (and it took me awhile to see it even after it was flagged).

I just pushed updates to further remove the useless `foundVertexState`, which hopefully makes this clearer. Also removed the commented out code.

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/internal/AbstractElement.java [hunk 2]
```diff
@@ -67,7 +67,7 @@ public boolean equals(Object other) {
             return getCompareId()==((AbstractElement)other).getCompareId();
         } else if (other instanceof JanusGraphElement) {
             return ((JanusGraphElement) other).hasId() && getCompareId()==((JanusGraphElement)other).longId();
-        } else if (other instanceof Element) {
+        } else if (other != null) {
```
Comment (mbrukman) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/internal/AbstractElement.java]: Why this change from `if (other instanceof Element)` to `if (other != null)`?

The new check is always true, because the first line checks for `other == null`; however, the next line casts `other` to `Element`, so we really should have checked whether `other instanceof Element` is true.

Should this line be reverted to the original?

File: janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java [hunk 3]
```diff
@@ -121,8 +121,7 @@
     public LuceneIndex(Configuration config) {
         String dir = config.get(GraphDatabaseConfiguration.INDEX_DIRECTORY);
         File directory = new File(dir);
-        if (!directory.exists()) directory.mkdirs();
-        if (!directory.exists() || !directory.isDirectory() || !directory.canWrite())
+        if ((!directory.exists() && !directory.mkdirs()) || !directory.isDirectory() || !directory.canWrite())
```
Comment (mbrukman) [inline janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java]: Since you're here, would you mind enclosing the `if` statement's single-line body in `{ ... }` so that it's clear that it refers to a single statement, please? We've had other situations in this code base where indentation was counter to the actual syntax; would be nice to avoid that going forward.

Incidentally, I like how Go enforces braces for `if` statements, even for a single line, to avoid this ambiguity.

File: janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java [hunk 4]
```diff
@@ -133,8 +132,7 @@ private Directory getStoreDirectory(String store) throws BackendException {
         String dir = basePath + File.separator + store;
         try {
             File path = new File(dir);
-            if (!path.exists()) path.mkdirs();
-            if (!path.exists() || !path.isDirectory() || !path.canWrite())
+            if ((!path.exists() && !path.mkdirs()) || !path.isDirectory() || !path.canWrite())
```
Comment (mbrukman) [inline janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java]: Please add `{...}` around `if`'s statement body.

File: janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrResultIterator.java [hunk 5]
```diff
@@ -63,7 +63,7 @@ public SolrResultIterator(SolrClient solrClient, Integer limit, int offset, int
 
     @Override
     public boolean hasNext() {
-        if (count != 0 && count % batchSize == 0 && batchSize * numBatches < limit) {
+        if (count != 0 && count % batchSize == 0 && count < limit) {
```
Comment (mbrukman) [inline janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrResultIterator.java]: This change _seems_ reasonable, but was this a Coverity-specified error or a separate change? I had to convince myself that this is correct, but this is a behavior change, so I'm curious how none of the unit tests failed before with this, and none required to be changed given this change (i.e., it should be possible to reproduce this error in a unittest such that it fails before, and succeeds after this change; or, that it succeeds before with an erroneous assertion, and this change requires changing said assertion to be correct).

Should this be a separate change?

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/internal/AbstractElement.java (hunk 2 above)
Comment (sjudeng) [reply janusgraph-core/src/main/java/org/janusgraph/graphdb/internal/AbstractElement.java]: @mbrukman At this point other is an instance of Element since otherwise it wouldn't pass the check [here](https://github.com/JanusGraph/janusgraph/blob/bfa3c970555932d4983155168322f72297aaee85/janusgraph-core/src/main/java/org/janusgraph/graphdb/internal/AbstractElement.java#L61-L63). But I did just push an update to remove the null check since you're right null was checked above.

File: janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrResultIterator.java (hunk 5 above)
Comment (sjudeng) [reply janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrResultIterator.java]: @mbrukman The Coverity finding was that `batchSize*numBatches` is performed at int32 but limit is int64. Upon review though I realized when `count % batchSize == 0` (e.g. at the start of a new batch), `count` is equal to `batchSize * numBatches`. I was familiar with this line from reviewing when it was added (see https://github.com/JanusGraph/janusgraph/pull/481#discussion_r136208445). @davidclement90 Does this seem okay?

Comment (mbrukman) [reply janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrResultIterator.java]: Agreed, LGTM.

Comment (davidclement90) [reply janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrResultIterator.java]: Yes it's ok LGTM
